### PR TITLE
UAVCAN Parameter Management & Enumeration

### PR DESCRIPTION
**Pull request by LorenzMeier**
@bendyer I've put in this PR for more formal code review and merging. It would be great if we could get the ROMFS-based firmware update in here as well.

### CODE REVIEW

**Comment by bendyer** [comment]
@LorenzMeier ROMFS-based firmware update is in 65c7872. The implementation copies all firmware files from the `ROMFS/uavcan/fw/…` directory into the appropriate location on the SD card, replacing previous firmware images unpacked from ROMFS, but not replacing any files the user has put there manually.

The firmware updater then runs as normal from the SD card.

This approach ensures that the user's firmware always overrides firmware in ROMFS, and also makes it easier for the user to see which firmware they're currently using when copying files to the SD card.

**Comment by LorenzMeier** [comment]
Awesome! Could you write a few sentences here or point me to a page detailing how to build and copy the ESC Firmware (the "naive" variant with the normal throttle response) for the SV ESC and for the Pixhawk ESC into the ROMFS? I'll probably go as far as integrating it in Travis and just ship the default master build with it.

File: src/modules/uavcan/uavcan_servers.hpp [hunk 1]
```diff
@@ -88,7 +95,7 @@ class UavcanServers
 	static constexpr unsigned QueuePoolSize =
 		(NumIfaces * uavcan::MemPoolBlockSize * MaxCanFramesPerTransfer);
 
-	static constexpr unsigned StackSize  = 3500;
+	static constexpr unsigned StackSize  = 6000;
```
Comment (LorenzMeier) [inline src/modules/uavcan/uavcan_servers.hpp]: Could you please rebase on master (I enabled -Os) and re-check stack sizes? We're just about out of RAM, so the few K here matter a lot.

Comment (LorenzMeier) [inline src/modules/uavcan/uavcan_servers.hpp]: Also if you could figure out how to build the lib -Os that would be much appreciated.

Comment (pavel-kirienko) [inline src/modules/uavcan/uavcan_servers.hpp]: A drastic change is coming up: https://github.com/UAVCAN/libuavcan/compare/footprint_reduction, ETA 2 days.

**Comment by LorenzMeier** [comment]
Awesome Pavel!

**Comment by LorenzMeier** [comment]
@bendyer Many thanks! Now we're missing only documentation how to get Firmware in and run stuff. A PR against https://github.com/PX4/Devguide which renders to http://dev.px4.io would be fantastic!

**Comment by bendyer** [comment]
@LorenzMeier will do.

File: src/modules/uavcan/uavcan_servers.hpp (hunk 1 above)
Comment (bendyer) [inline src/modules/uavcan/uavcan_servers.hpp]: Based on my tests a minimum of 4500 bytes is required. I'll check again once @pavel-kirienko's changes have landed and submit a PR with the actual size required.

**Pull request merged**